### PR TITLE
jbide-25733: fix jbosstools-openshift for Java 9

### PR DESCRIPTION
**Pull request by bdshadow**

### CODE REVIEW

**Comment by adietish** [comment]
@bdshadow dare to address the unneeded imports please?

**Comment by bdshadow** [comment]
@adietish  done already

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
@bdshadow thanks!

**Comment by adietish** [comment]
@bdshadow running the PR with java8 produces quite some errors, is this expected?
```Tests run: 491, Failures: 0, Errors: 268, Skipped: 4```

**Comment by adietish** [comment]
testPR

**Comment by bdshadow** [comment]
@adietish no, it is not expected. Everything must be fine, as the the last build: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/jbosstools-openshift-Pull-Request/1432/

i have no problems locally for both java 8 and java 9

**Comment by adietish** [comment]
@bdshadow I dont unterstand it, I see green jenkins but I cant make it work locally. Neither with java8 nor java9, in both cases I have 268 failing tests. Most with the following:
```
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.004 sec <<< FAILURE! - in org.jboss.tools.openshift.test.common.ui.OpenShiftCommonUIActivatorTest
testShowExplorerOnNewConnections(org.jboss.tools.openshift.test.common.ui.OpenShiftCommonUIActivatorTest)  Time elapsed: 0.002 sec  <<< ERROR!
org.mockito.exceptions.base.MockitoException: At the moment you cannot provide own implementations of MockHandler.
Please see the javadocs for the MockMaker interface.
	at org.powermock.api.mockito.repackaged.CglibMockMaker.cast(CglibMockMaker.java:31)
	at org.powermock.api.mockito.repackaged.CglibMockMaker.createMock(CglibMockMaker.java:23)
	at org.powermock.api.mockito.internal.mockmaker.PowerMockMaker.createMock(PowerMockMaker.java:47)
	at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:26)
	at org.mockito.internal.MockitoCore.mock(MockitoCore.java:51)
	at org.mockito.Mockito.mock(Mockito.java:1243)
	at org.mockito.Mockito.mock(Mockito.java:1120)
	at org.jboss.tools.openshift.test.common.ui.OpenShiftCommonUIActivatorTest.testShowExplorerOnNewConnections(OpenShiftCommonUIActivatorTest.java:58)
```
any idea?

**Comment by adietish** [comment]
@robstryker does this PR run without errors for you (like it does in jenkins)?

**Comment by bdshadow** [comment]
@adietish try to delete the lib directory:
```
[dbocharo@dbocharo jbosstools-openshift]$ cd tests/org.jboss.tools.openshift.test/
[dbocharo@dbocharo org.jboss.tools.openshift.test]$ pwd
/home/dbocharo/dev/openshift/jbosstools-openshift/tests/org.jboss.tools.openshift.test
[dbocharo@dbocharo org.jboss.tools.openshift.test]$ rm -rf lib
```

**Comment by adietish** [comment]
@bdshadow removing lib folder and getting rid of entries referring to lib/powermock* in org.jboss.tools.test/META-INF/MANIFEST.MF helped

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/WatchManager.java [hunk 1]
```diff
@@ -332,4 +332,11 @@ public void connectionRemoved(IConnection connection) {
 		}
 
 	}
+	
+	/**
+	 * Use it for test purposes only
+	 */
+	public Map<?,?> _getWatches() {
```
[the remark this answers is not in the document]
Comment (adietish) [reply plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/WatchManager.java]: this is the hack that we need in order to get around using powermock. So no, Sonar glad you point out, but you're wrong ;)

**Pull request merged**